### ISSUE TITLE
How set previous value of pattern in helm-occur?

### ISSUE THREAD

**Issue by lmdic** [emacs-helm/helm#2099]
`M-x helm-occur`
I input text "test" as pattern. As result helm-occur show line that contains "test". Nice. It's work fine.
But if I again M-x helm-occur I need again input some **pattern**. But very often I need to input same pattern. The question is: Is it possible to show previous **pattern**'s value when execute M-x `helm-occur` ?

**Comment by thierryvolpiatto** [comment]
Use `M-p`.

**Comment by lmdic** [comment]
> Use `M-p`.

`M-p` is a already taken
From Emacs tutorial:

`M-p `runs the command comint-previous-input (found in shell-mode-map),
which is an interactive compiled Lisp function in ‘comint.el’.

It is bound to <C-up>, M-p, <menu-bar> <inout> <previous-history>.

(`comint-previous-input ARG`)

**Comment by thierryvolpiatto** [comment]
No, you are wrong, please try to learn how keymaps are working in emacs.
When you run M-x helm-occur, you are certainly not using shell-mode-map.

Note that when using M-p (previous-history-element) only the used pattern are recorded in history, IOW if you quit helm-occur (C-g) the pattern will NOT be recorded in history.

**Comment by thierryvolpiatto** [comment]
Also you can use `helm-resume` to resume previous helm-occur session with same pattern.

**Comment by lmdic** [comment]
Here my test:

1. Open text file
2. `M-x helm-occur`, input patter: `java`
3. Result are 30 candidates
4. `C-g`
5. `M-x helm-occur`, input patter: `dalvik`
6. Result are 10 candidates
7.`C-g`
8. `M-x helm-occur`, input patter: `temp`
9. Result are 5 candidates
9.`C-g`
10. Go to begin on the file
11. `M-x count-matches` input `device` . Result 32 occurences
12. `M-x helm-occur` and pattern **is empty**

But I need to be set prev input value. In my case this is a `temp`.
Also I need to see all pattern's input in helm-occur and can select desire (previous input) pattern interactively.
Is it possible?

**Comment by thierryvolpiatto** [comment]
As I said, if you quit (C-g) pattern is not recorded in history.
So if you hit RET instead of C-g in 4) 7) and 9), you will have your three patterns recorded in history and you will be able to cycle these patterns with M-p.

**Comment by lmdic** [comment]
> As I said, if you quit (C-g) pattern is not recorded in history.
> So if you hit RET instead of C-g in 4) 7) and 9), you will have your three patterns recorded in history and you will be able to cycle these patterns with M-p.

Yes with `RET `is work... but I if I press `RET `the cursor go to specific line (when occur text). But I don't need this. I want to see only the result of `helm-occur`. So press `RET `is not very comfortable.
